include/aws/crt/iot/DeviceDefender.h

```diff
@@ -0,0 +1,184 @@
+#pragma once
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/crt/io/EventLoopGroup.h>
+#include <aws/crt/mqtt/MqttClient.h>
+
+#include <aws/iotdevice/device_defender.h>
+
+namespace Aws
+{
+    namespace Crt
+    {
+
+        namespace Io
+        {
+            class EventLoopGroup;
+        }
+
+        namespace Mqtt
+        {
+            class MqttConnection;
+        }
+
+        namespace Iot
+        {
+
+            /**
+             * Invoked upon DeviceDefender V1 task cancellation.
+             */
+            using OnDefenderV1TaskCancelledHandler = std::function<void(void *)>;
+
+            using DeviceDefenderReportFormat = aws_iotdevice_defender_report_format;
+
+            /**
+             * Enum used to expose the status of a DeviceDefenderV1 task.
+             */
+            enum class DeviceDefenderV1ReportTaskStatus
+            {
+                Ready = 0,
+                Running = 1,
+                Stopped = 2,
+                Failed = 3,
+            };
+
+            /**
+             * Represents a unique configuration for a device defender V1 report task. You can use a single instance of
+             * this class PER task you want to create.
+             */
+            class AWS_CRT_CPP_API DeviceDefenderV1ReportTaskConfig final
+            {
+
+                friend class DeviceDefenderV1ReportTask;
+                friend class DeviceDefenderV1ReportTaskConfigBuilder;
+
+              public:
+                /**
+                 * Creates a client configuration for use with making new AWS Iot specific MQTT Connections with MTLS.
+                 */
+                DeviceDefenderV1ReportTaskConfig(
```

Comment:
This could be private since there is a builder, unless we expect customer to use this constructor directly.

Reply:
Good point, with the Builder class refactored to create an actual DD Task instead of just a Config class, I think it makes sense to make the DDTask constructor private so only the Builder calls it.